prime-router/src/test/kotlin/fhirengine/engine/fhirRouterTests/FHIRBundleHelpersTests.kt

```diff
@@ -0,0 +1,441 @@
+
+        // assert
+        val provenance = bundle.entry.first { it.resource.resourceType.name == "Provenance" }.resource as Provenance
+        val outs = provenance.target
+        val receiversOut = outs.map { (it.resource as Endpoint).identifier[0].value }
+        assert(receiversOut.isNotEmpty())
+        assert(receiversOut[0] == "co-phd.full-elr-hl7")
+    }
+
+    @Test
+    fun `Test retrieving references for a Bundle resource`() {
+        val actionLogger = ActionLogger()
+        val fhirBundle = File("src/test/resources/fhirengine/engine/valid_data.fhir").readText()
+        val messages = FhirTranscoder.getBundles(fhirBundle, actionLogger)
+        val bundle = messages[0]
+
+        val diagnosticReport = FhirPathUtils.evaluate(
+            null,
+            bundle,
+            bundle,
+            "Bundle.entry.resource.ofType(DiagnosticReport)[0]"
+        )[0]
+
+        val diagnosticReportReferences = diagnosticReport.getResourceReferences()
+
+        assertThat(diagnosticReportReferences.count()).isEqualTo(5)
+    }
+
+    @Test
+    fun `Test retrieving properties for a Bundle resource`() {
+        val actionLogger = ActionLogger()
+        val fhirBundle = File("src/test/resources/fhirengine/engine/valid_data.fhir").readText()
+        val messages = FhirTranscoder.getBundles(fhirBundle, actionLogger)
+        val bundle = messages[0]
+
+        val diagnosticReport = FhirPathUtils.evaluate(
+            null,
+            bundle,
+            bundle,
+            "Bundle.entry.resource.ofType(DiagnosticReport)[0]"
+        )[0]
+
+        assertThat(diagnosticReport.getResourceProperties()).isNotEmpty()
+    }
+
+    @Test
+    fun `Test find Diagnostic report no observation`() {
+        val actionLogger = ActionLogger()
+        val fhirBundle =
+            File("src/test/resources/fhirengine/engine/bundle_diagnostic_report_no_observations.fhir").readText()
+        val messages = FhirTranscoder.getBundles(fhirBundle, actionLogger)
+        val bundle = messages[0]
+
+        assertThat(FHIRBundleHelpers.getDiagnosticReportNoObservations(bundle).count()).isEqualTo(1)
+    }
+
+    @Test
+    fun `Test Removing observation from diagnostic report with single observation`() {
+        val actionLogger = ActionLogger()
+        val fhirBundle = File("src/test/resources/fhirengine/engine/valid_data.fhir").readText()
+        val messages = FhirTranscoder.getBundles(fhirBundle, actionLogger)
+        val bundle = messages[0]
+
+        val diagnosticReport = FhirPathUtils.evaluate(
+            null,
+            bundle,
+            bundle,
+            "Bundle.entry.resource.ofType(DiagnosticReport)[0]"
+        )[0]
+
+        val observation = FhirPathUtils.evaluate(
+            CustomContext(bundle, bundle),
+            diagnosticReport,
+            bundle,
+            "%resource.result[0].resolve()"
+        )[0]
+
+        val observationReferences = observation.getResourceReferences()
```

Comment:
Add another assert to make sure you did get an observation and Diagnostic report here